src/thor/route_action.cc

```diff
@@ -80,6 +80,42 @@ void via_discontinuity(
   }
 }
 
+inline bool is_through_point(const valhalla::Location& l) {
+  return l.type() == valhalla::Location::kThrough || l.type() == valhalla::Location::kBreakThrough;
+}
+
+inline bool is_break_point(const valhalla::Location& l) {
+  return l.type() == valhalla::Location::kBreak || l.type() == valhalla::Location::kBreakThrough;
+}
+
+inline bool is_high_reachable(const valhalla::Location& loc,
```

Comment:
to avoid mistakes, maybe instead of sending the path edge to this function perhaps send the index and get the path edge from the passed in location or at least validate that this location owns this path edge

Reply:
`Location` has two vectors: `path_edges` and `filtered_edges`. So , if we send an index it will be not clear with edges to use

Reply:
I'd leave it as is

Reply:
ah! i hadnt noticed that you did this to the filtered edges as well. i guess worrying about accidentally passing the wrong edge location combo isnt so likely so we probably dont need an `assert` or something